### DISCUSSION TITLE
Creazione sensore che si la differenza di due

### DISCUSSION THREAD

**Discussion by gdr1993** [indomus/forum#3243] [HUB: Home Assistant]
Buongiorno vorrei un aiuto nella creazione di un sensore che sia la differenza di altri due. Questi sensori misurano la potenza istantanea di due dispositivi, vorrei creare un sensore che misuri appunto la differenza. purtroppo dalla sezione entità aiutanti il sistema permette la creazione solo di un sensore SOMMA ma non Differenza.
grazie in anticipo per l'aiuto

**Comment by indomus** [comment]
usa un sensore template https://indomus.it/formazione/esercizi-di-domotica-su-home-assistant-1-creare-dei-sensori-template/

**Comment by gdr1993** [comment]
Buonasera, grazie per le info, sto provando come da codice sotto ma non riesco a capire come mai non riesco a pescare il valore da sensor.controllo_assorbimento_clima_channel_1_power mi restituisce sempre 0. una volta capito come prendere il valore faro l'operazione di differenza, spero in qualche aiuto. grazie mille

`sensor: - platform: template sensors: clima_rimanente: device_class: power friendly_name: "Clima rimanente" value_template: '{{ states.sensor.controllo_assorbimento_clima_channel_1_power | float(default=0) }}' unit_of_measurement: "W"`

**Comment by gdr1993** [comment]
sono riuscito a sistemare il codice, come faccio se volessi arrotondare il risultato finale ad una sola cifra decimale? questo è il codice funzionante 

`sensor:
  - platform: template
    sensors:
      clima_rimanente:
        friendly_name: "Clima rimanente"
        unit_of_measurement: 'W'
        value_template: "{{ ((states('sensor.controllo_assorbimento_clima_channel_1_power') | float) - (states('sensor.controllo_assorbimento_clima_channel_2_power') | float)) - (states('sensor.shellyplusplugs_64b7080c7ea8_switch_0_power') | float) }}"`

**Reply by KC-inDomus** [reply]
devi usare la funzione round, cerca su google
